src/xrpld/ledger/detail/View.cpp

```diff
@@ -1353,6 +1353,18 @@ rippleSend(
 
     if (uSenderID == issuer || uReceiverID == issuer || issuer == noAccount())
     {
+        // if sender is issuer, check that the new OutstandingAmount will not
+        // exceed MaximumAmount
+        if (uSenderID == issuer)
+        {
+            auto const mptID = keylet::mptIssuance(saAmount.issue().getMptID());
+            auto const sle = view.peek(mptID);
+
+            if (sle->getFieldU64(sfOutstandingAmount) + saAmount.value() >
+                (*sle)[~sfMaximumAmount].value_or(maxMPTokenAmount))
+                return tecMPT_MAX_AMOUNT_EXCEEDED;
```

Comment:
Should add a unit-test for this.

Reply:
we already have existing unit tests that tests if the issuer issues out more than the max. the unit test i added covered the scenario reproduced in QA that currently exhibits the wrong behavior

Reply:
Isn't this a different execution path?

Reply:
All the existing test cases hit this new if-check in `rippleSend`. The problem found is that during holder to holder payment, `rippleCredit` is called twice from **1) holder to issuer** and then **2)issuer to holder**.  The current problem is that in the second `rippleCredit` call from issuer to holder, the code think the issuer is trying to issue more than the MaximumAmount. This is why I moved the check in the `rippleCredit` to its caller `rippleSend`.